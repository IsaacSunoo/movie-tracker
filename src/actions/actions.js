export const getMovies = (movies) => ({
    type: 'GET_MOVIES',
    movies
});

export const toggleFavorite = (id) => ({
  type: 'TOGGLE_FAVORITE',
  id
})

export const loginUser = (userInfo) => ({
  type: 'LOGIN_USER',
  userInfo
})