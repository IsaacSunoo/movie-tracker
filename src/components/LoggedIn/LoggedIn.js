import React, { Component } from 'react';
import { connect } from 'react-redux';

class LoggedIn extends Component {

  logout = () => {
    if (this.props.loginUser) {
      console.log(this.props.loginUser);
    }
  }

  displayFavorites = () => {
    
  }

    render() {
      const { name } = this.props.loginUser;
      console.log(this.props);
    return (
      <div className="user-display">
        <div className="user-welcome">Welcome {name}!</div>
        <div className="user-options">
          <span >Favorites</span> | <span className='logout-btn' onClick={this.logout}>Logout</span>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
    loginUser: state.loginUser
});

export default connect(mapStateToProps, null)(LoggedIn);
// export default LoggedIn;