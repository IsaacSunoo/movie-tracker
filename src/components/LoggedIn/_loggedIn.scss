.user-display {
    margin-right: 40px;
}

.user-welcome {
    color: #FFF;
    font-weight: bold;
    letter-spacing: 2px;
    margin-bottom: 20px;

}

.logout-btn {
    cursor: pointer;
}