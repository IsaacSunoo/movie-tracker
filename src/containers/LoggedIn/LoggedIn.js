import React, { Component } from 'react';
import { connect } from 'react-redux';
import { logoutUser } from '../../actions';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class LoggedIn extends Component {

  logout = () => {
    const { logoutUser, switchDisplay } = this.props;
    if (this.props.loginUser) {
      logoutUser();
      switchDisplay('login');
    }
  }

  render() {
      const { name } = this.props.loginUser;
    return (
      <div className="user-display">
        <div className="user-welcome">Welcome {name}!</div>
        <div className="user-options">
          <Link to='/favorites' className='favorites-link'><span className='display-favorites-btn'>Favorites &nbsp;</span></Link> | <Link to='/' className='logout-link'><span className='logout-btn' onClick={this.logout}>&nbsp; Logout</span></Link>
        </div>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
    loginUser: state.loginUser,
    favorites: state.favorites
});

const mapDispatchToProps = dispatch => ({
  logoutUser: () => dispatch(logoutUser())
});

LoggedIn.propTypes = {
  loginUser: PropTypes.object.isRequired,
  favorites: PropTypes.array.isRequired,
  logoutUser: PropTypes.func.isRequired
}

export default connect(mapStateToProps, mapDispatchToProps)(LoggedIn);