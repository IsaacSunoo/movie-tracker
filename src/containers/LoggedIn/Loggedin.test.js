import React from 'react';
import { LoggedIn, mapStateToProps, mapDispatchToProps } from './LoggedIn';
import { shallow } from 'enzyme';
import * as actions from '../../actions'

describe('LoggedIn', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <LoggedIn 
        loginUser= {name}
      />
    )
  });
  
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should add any stored favorites upon logging in', () => {

  });

  it('should getFavoriteMovies', () => {

  });

  it('should addStoredFavorites', () => {

  });

  it('should logout', () => {

  });

  describe('mapStateToProps', () => {
    it('should return an object with currentuser and favorites', () => {
      const mockState = {
        loginUser: {name: 'Isaac'},
        favorites: [123, 444],
        movies: [{title: 'Hey'}, {title: 'anotherMovie'}]
      }
      const expected = {
        loginUser: {name: 'Isaac'},
        favorites: [123, 444]
      }
      const result = mapStateToProps(mockState);
      expect(result).toEqual(expected)
    });
  });
  describe('mapDispatchToProps', () => {
    it('should call dispatch when using logoutUser', () => {
      const mockDispatch = jest.fn();
      const action = actions.logoutUser({name: 'Mason'})
      const mappedDispatch = mapDispatchToProps(mockDispatch)
      mappedDispatch.logoutUser({name: 'Mason'})
      expect(mockDispatch).toHaveBeenCalledWith(action)
    });
  });
});