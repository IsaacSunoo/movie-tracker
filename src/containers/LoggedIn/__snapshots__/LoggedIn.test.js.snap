// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedIn should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoggedIn
    loginUser="nodejs"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="user-welcome"
        >
          Welcome 
          !
        </div>,
        <div
          className="user-options"
        >
          <Link
            className="favorites-link"
            to="/favorites"
          >
            <span
              className="display-favorites-btn"
            >
              Favorites  
            </span>
          </Link>
           | 
          <Link
            className="logout-link"
            to="/"
          >
            <span
              className="logout-btn"
              onClick={[Function]}
            >
                Logout
            </span>
          </Link>
        </div>,
      ],
      "className": "user-display",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Welcome ",
            undefined,
            "!",
          ],
          "className": "user-welcome",
        },
        "ref": null,
        "rendered": Array [
          "Welcome ",
          undefined,
          "!",
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Link
              className="favorites-link"
              to="/favorites"
            >
              <span
                className="display-favorites-btn"
              >
                Favorites  
              </span>
            </Link>,
            " | ",
            <Link
              className="logout-link"
              to="/"
            >
              <span
                className="logout-btn"
                onClick={[Function]}
              >
                  Logout
              </span>
            </Link>,
          ],
          "className": "user-options",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span
                className="display-favorites-btn"
              >
                Favorites  
              </span>,
              "className": "favorites-link",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Favorites  ",
                "className": "display-favorites-btn",
              },
              "ref": null,
              "rendered": "Favorites  ",
              "type": "span",
            },
            "type": [Function],
          },
          " | ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span
                className="logout-btn"
                onClick={[Function]}
              >
                  Logout
              </span>,
              "className": "logout-link",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "  Logout",
                "className": "logout-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "  Logout",
              "type": "span",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="user-welcome"
          >
            Welcome 
            !
          </div>,
          <div
            className="user-options"
          >
            <Link
              className="favorites-link"
              to="/favorites"
            >
              <span
                className="display-favorites-btn"
              >
                Favorites  
              </span>
            </Link>
             | 
            <Link
              className="logout-link"
              to="/"
            >
              <span
                className="logout-btn"
                onClick={[Function]}
              >
                  Logout
              </span>
            </Link>
          </div>,
        ],
        "className": "user-display",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Welcome ",
              undefined,
              "!",
            ],
            "className": "user-welcome",
          },
          "ref": null,
          "rendered": Array [
            "Welcome ",
            undefined,
            "!",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Link
                className="favorites-link"
                to="/favorites"
              >
                <span
                  className="display-favorites-btn"
                >
                  Favorites  
                </span>
              </Link>,
              " | ",
              <Link
                className="logout-link"
                to="/"
              >
                <span
                  className="logout-btn"
                  onClick={[Function]}
                >
                    Logout
                </span>
              </Link>,
            ],
            "className": "user-options",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span
                  className="display-favorites-btn"
                >
                  Favorites  
                </span>,
                "className": "favorites-link",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Favorites  ",
                  "className": "display-favorites-btn",
                },
                "ref": null,
                "rendered": "Favorites  ",
                "type": "span",
              },
              "type": [Function],
            },
            " | ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <span
                  className="logout-btn"
                  onClick={[Function]}
                >
                    Logout
                </span>,
                "className": "logout-link",
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "  Logout",
                  "className": "logout-btn",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "  Logout",
                "type": "span",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
